test/services/RPCSelector.test.ts

```diff
@@ -19,25 +21,111 @@ describe('RPCSelector', () => {
   describe('#apply', () => {
     describe('when there is a single URL', () => {
       it('returns the preferred URL', async () => {
-        const url = 'http://127.0.0.1:8545'; 
+        const url = 'http://127.0.0.1:8545';
         const rpcSelector = new RPCSelector(url);
         expect(await rpcSelector.apply()).to.eq(url);
       });
     });
 
-    describe('when the preferred URL is up to date', () => {
-      it('returns the preferred URL', async () => {
-        const urls = ['https://data-seed-prebsc-1-s1.binance.org:8545/','https://data-seed-prebsc-1-s3.binance.org:8545/'];
-        const rpcSelector = new RPCSelector(urls);
-        expect(await rpcSelector.apply()).to.eq(urls[0]);
+    describe('when there are multiple URLs', () => {
+      describe('when the preferred URL is up to date', () => {
+        const stubValue = {
+          timestamp: Math.floor(Date.now() / 1000) - 30,
+        } as Block;
+
+        it('returns the preferred URL', async () => {
+          const mockedProvider = sinon.stub(JsonRpcProvider.prototype, 'getBlock').resolves(stubValue);
```

Comment:
this mock mocks for both URL calls, right?

Reply:
It only needs to mock one. The other one will not execute when the timestamp is up-to-date.

Reply:
We call `getBlock` on the first one and `getBlockNumber` on the rest of the URLs.